extensions/amp-story-subscriptions/0.1/test/test-amp-story-subscriptions.js

```diff
@@ -22,33 +30,113 @@ describes.realWin(
   (env) => {
     let win;
     let doc;
+    let storyEl;
+    let story;
     let subscriptionsEl;
     let storeService;
     let storySubscriptions;
+    let subscriptionService;
+
+    /**
+     * @param {number} count
+     * @param {Array<string>=} ids
+     * @return {!Array<!Element>}
+     */
+    async function createStoryWithPages(count, ids = []) {
+      Array(count)
+        .fill(undefined)
+        .map((unused, i) => {
+          const page = (
+            <amp-story-page
+              auto-advance-after="2s"
+              id={ids && ids[i] ? ids[i] : `-page-${i}`}
+            ></amp-story-page>
+          );
+          storyEl.appendChild(page);
+        });
 
-    const nextTick = () => new Promise((resolve) => win.setTimeout(resolve, 0));
+      AmpStory.isBrowserSupported = () => true;
+      story = await storyEl.getImpl();
+    }
 
-    beforeEach(() => {
+    beforeEach(async () => {
       win = env.win;
       doc = win.document;
 
+      toggleExperiment(win, 'amp-story-paywall-exp', true);
```

Comment:
Please be accurate with the experiment name: it enabled amp-subscriptions, and doesn't need to be suffixed with `-exp` as no other experiments do it.

Reply:
In that case, how about 'amp-story-with-amp-subscriptions' or 'enable-amp-subscriptions-in-amp-story' or ''amp-story-amp-subscriptions'? Not sure about the canonical naming structure so making sure here.

Reply:
Changed name to 'enable-amp-story-subscriptions'